src/mappings/core.ts

```diff
@@ -481,9 +485,13 @@ export function handleSwap(event: Swap): void {
   )
 
   // update swap event
+  let tokenIn = amount1Out > BigDecimal.fromString('0') ? pair.token0 : pair.token1
```

Comment:
BigDecimal has `.gt` function for "greater than". I don't think using `>` works here, please check again

Reply:
Changed as requested